core/src/transfer/data.rs

```diff
@@ -296,3 +483,97 @@ impl ContractBytecode {
         Ok(Self { hash, bytes })
     }
 }
+
+impl BlobData {
+    /// Version of the KZG commitment hash used in versioned blob hashes.
+    pub const VERSIONED_HASH_VERSION_KZG: u8 = 0x01;
+
+    /// Provides contribution bytes for an external hash.
+    #[must_use]
+    pub fn to_hash_input_bytes(&self) -> Vec<u8> {
+        self.hash.to_vec()
+    }
+
+    /// Serializes this object into a variable length buffer
+    #[must_use]
+    pub fn to_var_bytes(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        bytes.extend(self.hash);
+        if let Some(data) = &self.data {
+            bytes.push(1u8);
+            bytes.extend(data.to_var_bytes());
+        } else {
+            bytes.push(0u8);
+        }
+        bytes
+    }
+
+    /// Deserializes from a bytes buffer.
+    /// Resets buffer to a position after the bytes read.
+    ///
+    /// # Errors
+    /// Errors when the bytes are not available.
+    pub fn from_buf(buf: &mut &[u8]) -> Result<Self, BytesError> {
+        let hash = crate::read_arr(buf)?;
+
+        let data = match u8::from_reader(buf)? {
+            0 => None,
+            1 => Some(BlobSidecar::from_buf(buf)?),
+            _ => return Err(BytesError::InvalidData),
+        };
+
+        Ok(Self { hash, data })
+    }
+
+    /// Take the data field, if it exists.
```

Comment:
not sure if this is necessary, but I think it's good to underline it

```suggestion
    /// Take the data field, if it exists.
    /// Note that `data` is replaced with `None`.
```

Reply:
That's superfluous, what [`take`](https://doc.rust-lang.org/std/option/enum.Option.html#method.take) does is usually well known